keyring/testing/backend_macOS.py

```diff
@@ -0,0 +1,173 @@
+        service = random_string(20, self.DIFFICULT_CHARS)
+        self.keyring.set_password(service, username1, password)
+        self.set_password(service, username2, password)
+        self.keyring.delete_password(service, username1)
+        assert self.keyring.get_password(service, username2) == password
+
+    def test_name_property(self):
+        assert is_ascii_printable(self.keyring.name)
+
+    def test_unicode_chars(self):
+        password = random_string(20, UNICODE_CHARS)
+        username = random_string(20, UNICODE_CHARS)
+        service = random_string(20, UNICODE_CHARS)
+        self.check_set_get(service, username, password)
+
+    def test_unicode_and_ascii_chars(self):
+        source = (
+            random_string(10, UNICODE_CHARS)
+            + random_string(10)
+            + random_string(10, self.DIFFICULT_CHARS)
+        )
+        password = random_string(20, source)
+        username = random_string(20, source)
+        service = random_string(20, source)
+        self.check_set_get(service, username, password)
+
+    def test_different_user(self):
+        """
+        Issue #47 reports that WinVault isn't storing passwords for
+        multiple users. This test exercises that test for each of the
+        backends.
+        """
+
+        keyring = self.keyring
+        self.set_password('service1', 'user1', 'password1')
+        self.set_password('service1', 'user2', 'password2')
+        assert keyring.get_password('service1', 'user1') == 'password1'
+        assert keyring.get_password('service1', 'user2') == 'password2'
+        self.set_password('service2', 'user3', 'password3')
+        assert keyring.get_password('service1', 'user1') == 'password1'
+
+    def test_credential(self):
+        keyring = self.keyring
+
+        cred = keyring.get_credential('service', None)
+        assert cred is None
+
+        self.set_password('service1', 'user1', 'password1')
+        self.set_password('service1', 'user2', 'password2')
+
+        cred = keyring.get_credential('service1', None)
+        assert cred is None or (cred.username, cred.password) in (
+            ('user1', 'password1'),
+            ('user2', 'password2'),
+        )
+
+        cred = keyring.get_credential('service1', 'user2')
+        assert cred is not None
+        assert (cred.username, cred.password) in (
+            ('user1', 'password1'),
+            ('user2', 'password2'),
+        )
+
+    def test_set_properties(self, monkeypatch):
+        env = dict(KEYRING_PROPERTY_FOO_BAR='fizz buzz', OTHER_SETTING='ignore me')
+        monkeypatch.setattr(os, 'environ', env)
+        self.keyring.set_properties_from_env()
+        assert self.keyring.foo_bar == 'fizz buzz'
+
+    def test_keychain_declaration(self):
+        self.keyring.set_password(
+            'service',
+            'user1',
+            'password1',
+            # '/Users/<insert local user here>/Library/Keychains/APIs.keychain-db',
```

Comment:
Any tests need to capture the desired behavior or be skipped. This test claims to test the keychain declaration, but in fact doesn't do any such thing, but instead has commented code.